apiary-receivers/apiary-receiver-sqs/src/main/java/com/expedia/apiary/extensions/receiver/sqs/messaging/SqsMessageReader.java

```diff
@@ -60,34 +68,51 @@ public void close() {
   }
 
   @Override
-  public Optional<ListenerEvent> read() {
+  public Optional<MessageEvent> read() {
     if (records == null || !records.hasNext()) {
       records = receiveMessage();
     }
     if (records.hasNext()) {
       Message message = records.next();
-      delete(message);
-      return Optional.of(eventPayLoad(message));
+      MessageEvent sqsMessageEvent = messageEvent(message);
+      return Optional.of(sqsMessageEvent);
     } else {
       return Optional.empty();
     }
   }
 
-  private void delete(Message message) {
+  public void delete(String receiptHandle) {
     DeleteMessageRequest request = new DeleteMessageRequest()
         .withQueueUrl(queueUrl)
-        .withReceiptHandle(message.getReceiptHandle());
+        .withReceiptHandle(receiptHandle);
     consumer.deleteMessage(request);
   }
 
+  public void extendVisibilityTimeout(String receiptHandle) {
```

Comment:
Do we need this? As long as SY has a single thread reading off the queue and it deletes on replication success before going back to the queue for more, even if the message has become visible in the interim the delete should remove it before its (re-)read.

Reply:
Given that the only consumer of this at the moment is Shunting Yard, which also reads messages in a single thread, I've removed this method.